src/bindings/c/tests/ov_multithreading_test.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <thread>
+
+#include "ov_test.hpp"
+
+class ov_multithreading_test : public ov_capi_test_base {
+    void SetUp() override {
+        ov_capi_test_base::SetUp();
+    }
+
+    void TearDown() override {
+        ov_capi_test_base::TearDown();
+    }
+
+public:
+    void run_parallel(std::function<void(void)> func,
+                      const unsigned int iterations = 100,
+                      const unsigned int threads_num = 8) {
+        std::vector<std::thread> threads(threads_num);
+
+        for (auto& thread : threads) {
+            thread = std::thread([&]() {
+                for (unsigned int i = 0; i < iterations; ++i) {
+                    func();
+                }
+            });
+        }
+
+        for (auto& thread : threads) {
+            if (thread.joinable())
+                thread.join();
+        }
+    }
+    std::vector<std::pair<std::string, std::string>> networks;
+    void set_up_networks(const unsigned int iterations = 100) {
+        for (unsigned i = 0; i < iterations; i++) {
+            std::pair<std::string, std::string> network(xml_file_name, bin_file_name);
+            networks.emplace_back(network);
+        }
+    }
+};
+
+INSTANTIATE_TEST_SUITE_P(ov_capi, ov_multithreading_test, ::testing::Values("CPU"));
+
+TEST_P(ov_multithreading_test, get_property) {
+    auto device_name = GetParam();
+    ov_core_t* core = nullptr;
+    OV_EXPECT_OK(ov_core_create(&core));
+    EXPECT_NE(nullptr, core);
+
+    ov_model_t* model = nullptr;
+    OV_EXPECT_OK(ov_core_read_model(core, xml_file_name.c_str(), bin_file_name.c_str(), &model));
+    EXPECT_NE(nullptr, model);
+
+    ov_compiled_model_t* compiled_model = nullptr;
+    OV_EXPECT_OK(ov_core_compile_model(core, model, device_name.c_str(), 0, &compiled_model));
+    EXPECT_NE(nullptr, compiled_model);
+
+    const char* key = ov_property_key_supported_properties;
+    char* result = nullptr;
+
+    run_parallel([&]() {
+        OV_EXPECT_OK(ov_compiled_model_get_property(compiled_model, key, &result));
```

Comment:
Memory leak is caused by this line. 
ov_compiled_model_get_property() will allocate array and return as result for each threads, but the result only be freed for one time.

Reply:
```suggestion
        OV_EXPECT_OK(ov_compiled_model_get_property(compiled_model, key, &result));
        ov_free(result);
```